contracts/SimpleSwap.sol

```diff
@@ -116,53 +97,39 @@ contract SimpleSwap is SW3Utils {
   /// @param value maximum amount to send
   /// @return payout amount that was actually paid out
   /// @return payout amount that bounced
-  function _payout(address beneficiary, uint value) internal returns (uint payout, uint bounced) {
-    /* part of hard deposit used */
-    payout = Math.min256(value, hardDeposits[beneficiary].amount);
-    /* if there some of the hard deposit is used update the structure */
-    if(payout != 0) {
-      hardDeposits[beneficiary].amount -= payout;
-      totalDeposit -= payout;
+  function _payout(address beneficiary, uint amount) internal returns (bool) {
+    /* SWAP contract should be allowed to process the payout */
+    if(amount >= liquidBalance()) {
+      return false; // returning false signals a bounced cheque. TODO: consider reverting
```

Comment:
We return false directly when the liquidBalance cannot hold the requested amount. Since a beneficiary can request a partial payout from his cheque balance, there is no need to transfer at this stage; a beneficiary can just submit a payout request with a lower value.